tools/mason/MasonBuild.chpl

```diff
@@ -222,25 +230,41 @@ proc compileSrc(lockFile: borrowed Toml, binLoc: string, show: bool,
    url and the name for local mason dependency pool */
 proc genSourceList(lockFile: borrowed Toml) {
   var sourceList: list((string, string, string));
+  var gitList: list((string, string, string));
   for (name, package) in lockFile.A.items() {
     if package!.tag == fieldtag.fieldToml {
       if name == "root" || name == "system" || name == "external" then continue;
       else {
         var toml = lockFile[name]!;
+        // TODO: What do we want to do with version for git deps?
         var version = toml["version"]!.s;
-        var source = toml["source"]!.s;
-        sourceList.append((source, name, version));
+
+        if toml.pathExists("source") {
+          var source = toml["source"]!.s;
+          sourceList.append((source, name, version));
+        } else if toml.pathExists("url") {
+          var url = toml["url"]!.s;
+
+          var branch: string;
+          // use branch if specified, else default to master
+          if toml["branch"] != nil {
+            branch = toml["branch"]!.s;
+          } else {
+            branch = "master";
```

Comment:
can\should we default to `main` here?

Reply:
As mentioned above, I think `master` is still default, so that's where I would go (the one user repository I was looking at had `master` still).

I do wonder now that you point it out if we should just be defaulting to `HEAD` though, so we wouldn't even pull a branch or anything in that case, just use whatever it gives us as the default.

Reply:
For my own understanding, does this mean that for a package dependency that is already in a local git repo, that mason would use `HEAD` of whichever branch I have checked out?

Reply:
No, that is not how this is setup to work today. You can't really point it at an existing git repo in this PR (that could be something to add, but isn't part of this effort), you can only specify a git repo in your toml and it will be cloned into `$MASON_HOME/git/` with the directory name `myRepo-branch`, where `branch` would be `HEAD` in this case, where I would expect it to default to whatever branch it is set to after cloning the repo. 

This is based off of Rusts implementation, I did not try to come up with new functionality on top of what Rust does for this initial effort

Reply:
Also, FWIW, I do not think that should be added to this PR, I think we should get this merged with the functionality that satisfies the user request (which it should have accomplished now) and then improve upon it as necessary. I don't think we should try to consider a bunch of additional cases when there are no existing uses of it today.

Reply:
> I do not think that should be added to this PR

I didn't mean to suggest anything be added, I was just looking to understand the comment here:

> ... if we should just be defaulting to HEAD though, so we wouldn't even pull a branch or anything in that case

which I think you clarified for me with this explanation about directory structure:

> it will be cloned into $MASON_HOME/git/ with the directory name `myRepo-branch`

Now I think I understand correctly: If a user specified a git repo dependency that used `working-branch`, and then later changed the dependency to be `main`, they would have two folders:

* `$MASON_HOME/git/myRepo-working-branch` 
* `$MASON_HOME/git/myRepo-main`

and even if the user did go inside one of those local repos and manually checked out the wrong branch, we would exec a `git checkout <branch>` command before building anyway.

Reply:
Yup, that's right, they would have multiple directories.